### PR TITLE
776 add question help

### PR DESCRIPTION
**Pull request by Ancentury**

### CODE REVIEW

**Comment by MrDeshaies** [comment]
The current button to show help is too prominent, it is too big and loud. Instead, it should be a small neutral icon so as to not attract undue attention.

image

An "i" in a circle is a popular tooltip/help icon. Could something like this be used instead? 
image

**Comment by Ancentury** [comment]
Changed to "i" icon. Please check again.

File: src/views/Home.vue [hunk 1]
```diff
@@ -144,6 +154,12 @@ export default class Home extends Vue {
             ' <strong class="required">(' + requiredText + ")</strong>";
         }
 
+        if (options.question.help) {
+          let helpTxt = sender.locale == "fr" ? String(options.question.help.fr) : String(options.question.help.default);
+          let showHelp = sender.locale == "fr" ? "Afficher l'aide" : "Show help";
```
Comment (MrDeshaies) [inline src/views/Home.vue:159]: Please externalize these strings to en.json and fr.json, using this.$t("key") to get the translation. This keeps all translations in one spot.

**Comment by MrDeshaies** [comment]
Found a strange bug with visibleIf. I added help text for most questions in section 9 about the data. When I answer "Yes" to the first question, which triggers other questions to become visible, the help text for the last question to become visible is applied to all other questions. 
[survey-enfr.json.txt](https://github.com/canada-ca/aia-eia-js/files/7873784/survey-enfr.json.txt)

**Steps to reproduce:**
1. Replace survey-enfr.json with the file above.
2. Start a new survey
3. On first page, select design stage
4. Navigate to section 9 "about the data"
5. Answer Yes to the first question (Will the Automated Decision System use personal information as input data?)
6. Notice the help text for all questions now displays the entry for the question about PIBs and PIAs (`"name": "aboutDataSource11"`)

**Comment by MrDeshaies** [comment]
> Changed to "i" icon. Please check again.

The icon looks good :)

**Comment by MrDeshaies** [comment]
There's a bug. When there are more than two questions in the same section with help text, all questions display the help for the last question. I think Home.vue, line 161, is the problem.

`document.getElementById("helpText")!.innerHTML = helpTxt;`

This should be done in the handler showHelp() defined in index.html, not as part of rendering the questions for the section in Home.vue.

File: src/views/Home.vue (hunk 1 above)
Comment (Ancentury) [reply src/views/Home.vue]: Fixed.

**Comment by Ancentury** [comment]
Nice catch. Fixed. Please check again.

File: src/views/Home.vue [hunk 2]
```diff
@@ -144,6 +154,19 @@ export default class Home extends Vue {
             ' <strong class="required">(' + requiredText + ")</strong>";
         }
 
+        if (options.question.help) {
+          let helpTxt = sender.locale == "fr" ? String(options.question.help.fr) : String(options.question.help.default);
+          helpTxt = helpTxt
+            .replace(/&/g, '&amp;')
+            .replace(/"/g, '&quot;')
+            .replace(/'/g, 'ooooo')
```
Comment (nschonni) [inline src/views/Home.vue:162]: Not sure this escape makes sense. I think this replace section should just be using something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Comment (Ancentury) [reply src/views/Home.vue:162]: Good eye. This escape is similar to encodeURIComponent, but add some more escape, and work around.

**Review by MrDeshaies — APPROVED** [review]

**Pull request merged**